src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -237,3 +272,36 @@ impl DeviceManager {
         Err(anyhow!("ID are exhausted"))
     }
 }
+
+// Many scenarios have similar steps when adding devices. so to reduce duplicated code,
+// we should create a common method abstracted and use it in various scenarios.
+// do_handle_device:
+// (1) new_device with DeviceConfig and return device_id;
+// (2) try_add_device with device_id and do really add device;
+// (3) return device info of device's info;
+pub async fn do_handle_device(
```

Comment:
I think you could implement this function for device manager, and make `new_device` and `try_add_device` private.

Reply:
Thx, fixed